.github/workflows/rust.yml

```
@@ -180,3 +180,17 @@ jobs:
       - name: Deploy develop # unused
         if: github.ref == 'refs/heads/develop'
         run: bash scripts/deploy_docker.sh develop
+
+  autodeploy:
+    runs-on: ubuntu-latest
+    needs: [deploy]
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/')
```

The following should be enough:

```suggestion
    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

This project only uses `main` for deployments to `staging`.

---

Updated.